example/x64/amd/default/src/start.s

```diff
@@ -0,0 +1,27 @@
+/// @copyright
+/// Copyright (C) 2020 Assured Information Security, Inc.
+///
+/// @copyright
+/// Permission is hereby granted, free of charge, to any person obtaining a copy
+/// of this software and associated documentation files (the "Software"), to deal
+/// in the Software without restriction, including without limitation the rights
+/// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+/// copies of the Software, and to permit persons to whom the Software is
+/// furnished to do so, subject to the following conditions:
+///
+/// @copyright
+/// The above copyright notice and this permission notice shall be included in
+/// all copies or substantial portions of the Software.
+///
+/// @copyright
+/// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+/// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+/// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+/// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+/// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+/// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+/// SOFTWARE.
+
+.global _start
+_start:
```

Comment:
Are you intending for every example/extension to define its own _start function? Thats usually something taken care of by a C runtime library, so I would think we would want to do the same pattern here (provide a _start in the runtime project, and let the extension/example define a main function)

Reply:
Yes and no. I removed it for now, but it might go back in at some point. Typically, the runtime is actually _start, and the crap around that and nothing more. What's there right now is the syscall logic, which is different. On most platforms, the syscall interface and the CRT are not in the same library. Bareflank currently has a syscall library, a pthread library and a CRT library. pthreads are no longer needed. It might make more sense to rename this to the syscall library, and then create another one specific to the CRT.

Reply:
I renamed it to "syscall" and added a "runtime" that has the start logic in it. I then modified the examples so that they link against both, and the examples no longer have the triple, or their own start.s. If we want to merge the syscall and runtime libraries into a single library in the future we can easily do that, but for now, it makes sense to keep them separate. So... this should be fixed.